packages/@lwc/engine-core/src/libs/mutation-tracker/index.ts

```diff
@@ -120,9 +129,17 @@ export class ReactiveObserver {
         this.callback.call(undefined, this);
     }
 
-    link(reactiveObservers: ReactiveObserver[]) {
-        ArrayPush.call(reactiveObservers, this);
+    // jtu-todo: add some comments here about why CallbackFunction is an acceptable type to link (eg. it's for signals)
+    // technically the CallbackFunction takes a ReactiveObserver argument but we're not passing one in with the subscribe
+    link(reactiveObservers: ReactiveObserver[] | CallbackFunction) {
+        if (Array.isArray(reactiveObservers)) {
```

Comment:
Overloading this method with two dissimilar inputs for the same argument seems not ideal. Would it be possible to add a separate `linkSignal` method specifically for the new use case?

Reply:
This has been moved out too!